File: test-data/loginData.js
module.exports = {
  validUser:    { username: 'sajith_xyz',  password: 'User@123' },
  invalidUser:  { username: 'wrong_user',  password: 'WrongPass@999' },
  emptyUser:    { username: '',            password: '' },
  boundaryData: {
    longUsername:    'a'.repeat(256),
    longPassword:    'P@ssword' + 'x'.repeat(248),
    sqlInjection:    "' OR 1=1 --",
    sqlInjectionPwd: "' OR '1'='1",
    xssPayload:      '<script>alert(\'xss\')</script>',
    specialChars:    '!@#$%^&*()',
    whitespaceOnly:  '   ',
  },
};

File: dotnet-tests/TestData/LoginData.vb
Public Class LoginCredential
    Public Property Username As String
    Public Property Password As String

    Public Sub New(username As String, password As String)
        Me.Username = username
        Me.Password = password
    End Sub
End Class

Public Module LoginData

    Public ReadOnly ValidUser As New LoginCredential("sajith_xyz", "User@123")
    Public ReadOnly InvalidUser As New LoginCredential("wrong_user", "WrongPass@999")

    Public ReadOnly LongUsername As String = New String("a"c, 256)
    Public ReadOnly LongPassword As String = "P@ssword" & New String("x"c, 248)
    Public Const SqlInjection As String = "' OR 1=1 --"
    Public Const SqlInjectionPwd As String = "' OR '1'='1"
    Public Const XssPayload As String = "<script>alert('xss')</script>"
    Public Const SpecialChars As String = "!@#$%^&*()"
    Public Const WhitespaceOnly As String = "   "

End Module
